src/codechandler/HEVCCodecHandler.cpp

```diff
@@ -0,0 +1,79 @@
+#include "HEVCCodecHandler.h"
+
+HEVCCodecHandler::HEVCCodecHandler(AP4_SampleDescription* sd) : CodecHandler(sd)
+{
+  if (AP4_HevcSampleDescription* hevc =
+          AP4_DYNAMIC_CAST(AP4_HevcSampleDescription, m_sampleDescription))
+  {
+    m_extraData.SetData(hevc->GetRawBytes().GetData(), hevc->GetRawBytes().GetDataSize());
+    m_naluLengthSize = hevc->GetNaluLengthSize();
+  }
+}
+
+bool HEVCCodecHandler::ExtraDataToAnnexB()
+{
+  if (AP4_HevcSampleDescription* hevc =
+          AP4_DYNAMIC_CAST(AP4_HevcSampleDescription, m_sampleDescription))
+  {
+    const AP4_Array<AP4_HvccAtom::Sequence>& sequences = hevc->GetSequences();
+
+    if (!sequences.ItemCount())
+    {
+      kodi::Log(ADDON_LOG_WARNING, "No available sequences for HEVC codec extra data");
+      return false;
+    }
+
+    //calculate the size for annexb
+    size_t sz(0);
+    for (const AP4_HvccAtom::Sequence *b(&sequences[0]), *e(&sequences[sequences.ItemCount()]);
```

Comment:
For here I'm thinking of rewriting a bit more succinctly:
 ```
for (unsigned int i{0}; i < sequences.ItemCount(); ++i)
      for (unsigned int j{0}; j < (&sequences[i])->m_Nalus.ItemCount(); ++i)
        sz += (4 + (&sequences[i])->m_Nalus[j].GetDataSize());
```
@CastagnaIT does this seem better to you?

Reply:
good thing better readable
please on `for` loops always add brackets,
relying on spaces in this case is not a good thing
a little space oversights can cause hidden problems